### PR TITLE
New tool to sync profile data from OpenAPS to Nightscout

### PR DESCRIPTION
**Pull request by sulkaharo**
DO NOT MERGE YET! Someone who uses mg/dL needs to verify this works.

Note this expands the data contained in profile.json to contain a full pump profile and rounds some of the contained numbers to get rid of excessive fractions.

### CODE REVIEW

**Comment by sulkaharo** [comment]
Add the following alias to openaps.ini to enable this

```
upload-profile = ! bash -c "oref0-upload-profile settings/profile.json $NIGHTSCOUT_HOST $API_SECRET"
```

and changed the following line

```
upload = ! bash -c "echo -n Upload && ( openaps upload-ns-status;  openaps upload-recent-treatments; openaps upload-profile ) 2>/dev/null >/dev/null && echo ed"
```

**Comment by sulkaharo** [comment]
I'm good with a merge but note this is still AFAIK untested by a mg/dL user. I have it on our production rig though and it seems to work.

File: lib/profile/index.js [hunk 1]
```diff
@@ -48,6 +49,11 @@ function generate (inputs, opts) {
 
   profile.current_basal = basal.basalLookup(inputs.basals);
   profile.basalprofile = inputs.basals;
+  
+  _.forEach(profile.basalprofile, function(basalentry) {
+    basalentry.rate = +(Math.round(basalentry.rate + "e+3")  + "e-3");
```
Comment (scottleibrand) [inline lib/profile/index.js]: What does this do?  Is it another way of rounding to three digits after the decimal place?

**Comment by scottleibrand** [comment]
Merging this to the autotune-mdi branch for now until we get some more people testing it.

**Comment by sulkaharo** [comment]
@oschumac said he's running this in a production rig now

**Comment by scottleibrand** [comment]
Great!  I think I'll leave it in autotune-mdi for now until someone wants to take on the task of integrating it with oref0-setup.

**Comment by CaptainBalou** [comment]
Normally I sync the profile with a simple cron job. But as I see some of you tried to integrate that in a better way. Cool!
I am not sure if this in dev now - cannot interpret the history here. Since I am using mg/dl I could give that a try. Can someone assist?

**Comment by scottleibrand** [comment]
Yeah, it should be in master by now too.

**Comment by CaptainBalou** [comment]
I am on dev. So that means I have to put the aliases in place and nothing else?

**Pull request merged**